src/compiler/scala/tools/reflect/ReflectGlobal.scala

```diff
@@ -11,6 +12,12 @@ import scala.tools.nsc.Settings
 class ReflectGlobal(currentSettings: Settings, reporter: Reporter, override val rootClassLoader: ClassLoader)
   extends Global(currentSettings, reporter) with scala.tools.reflect.ReflectSetup with scala.reflect.runtime.SymbolTable {
 
+  override lazy val analyzer = new {
+    val global: ReflectGlobal.this.type = ReflectGlobal.this
+  } with Analyzer {
+    override lazy val macroParentClassLoader = rootClassLoader
+  }
+
```

Comment:
Can we simply override `findMacroClassloader` here, just like `ReplGlobal` does? https://github.com/scala/scala/blob/1fbce4612c21a4d0c553ea489b4765494828c09f/src/repl/scala/tools/nsc/interpreter/ReplGlobal.scala#L27